prov/efa/src/rdm/efa_rdm_ep.h

```diff
@@ -304,11 +304,8 @@ bool efa_rdm_ep_should_write_rnr_completion(struct efa_rdm_ep *ep)
 static inline
 int efa_rdm_ep_use_p2p(struct efa_rdm_ep *efa_rdm_ep, struct efa_mr *efa_mr)
 {
-	if (!efa_mr)
-		return 0;
-
 	/*
-	 * always send from host buffers if we have a descriptor
+	 * always send from host buffers no matter efa_mr is registered or not
```

Comment:
This comment is not correct. We are not always sending from host buffers if efa_mr is NULL. For eager send, we will do a copy from host buffer to registered bounce buffer. 

We can just change it to

```
P2p is always available for host memory (Unregistered buffer will be regarded as host memory as EFA provider requires FI_MR_HMEM)
```